src/GenAPI/Microsoft.DotNet.GenAPI.Task/GenAPITask.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.Build.Tasks;
+using Microsoft.DotNet.GenAPI.Shared;
+
+namespace Microsoft.DotNet.GenAPI.Tasks;
+
+#nullable enable
+
+/// <summary>
+/// MSBuild task frontend for the Roslyn-based GenAPI.
+/// </summary>
+public class GenAPITask : BuildTask
+{
+    /// <summary>
+    /// Delimited (',' or ';') set of paths for assemblies or directories to get all assemblies.
+    /// </summary>
+    [Required]
+    public string? Assembly { get; set; }
+
+    /// <summary>
+    /// If true, tries to resolve assembly reference.
+    /// </summary>
+    public bool? ResolveAssemblyReferences { get; set; }
```

Comment:
Booleans very rarely need to be nullable. You don't need a third state here, false and true should be sufficient:

```suggestion
    public bool ResolveAssemblyReferences { get; set; }
```